[processor/transform] Add extract_sum_metric OTTL function (#24368)
open-telemetry/opentelemetry-collector-contrib: processor/transformprocessor/README.md

- `convert_gauge_to_sum("delta", true)`

> ### convert_histogram_sum_val_to_sum

Reviewer: Maybe `extract_sum_metric`? So it's more obvious what it's supposed to do.

> ### convert_histogram_sum_val_to_sum

Reviewer 2: Ya

- `convert_gauge_to_sum("delta", true)`

### extract_sum_metric

> **Note** This function supports Histograms, ExponentialHistograms and Summaries.

`extract_sum_metric(is_monotonic, aggregation_temporality)`

> The `extract_sum_metric` function creates a new Sum metric from a Histogram, ExponentialHistogram or Summary's sum value.

Reviewer 3: ```suggestion
The `extract_sum_metric` function creates a new Sum metric from a Histogram, ExponentialHistogram or Summary's sum value. If the sum value of a Histogram or ExponentialHistogram data point is missing, no data point is added to the output metric. A metric will only be created if there is at least one data point.
```

- `convert_gauge_to_sum("delta", true)`

### extract_sum_metric

> [!NOTE]  
> This function supports Histograms, ExponentialHistograms and Summaries.

> `extract_sum_metric(is_monotonic, aggregation_temporality)`

Reviewer 3: ```suggestion
`extract_sum_metric(is_monotonic)`
```

The `extract_sum_metric` function creates a new Sum metric from a Histogram, ExponentialHistogram or Summary's sum value. If the sum value of a Histogram or ExponentialHistogram data point is missing, no data point is added to the output metric. A metric will only be created if there is at least one data point.

`is_monotonic` is a boolean representing the monotonicity of the new metric.

> The name for the new metric will be `<original metric name>_sum`. The fields that are copied are: `timestamp`, `starttimestamp`, `attibutes`, and `description` `aggregation_temporality`. As metrics of type Summary don't have an `aggregation_temporality` attribute, this attribute will be set to `AGGREGATION_TEMPORALITY_CUMULATIVE` for those metrics.

Reviewer 3: ```suggestion
The name for the new metric will be `<original metric name>_sum`. The fields that are copied are: `timestamp`, `starttimestamp`, `attibutes`, `description`, and `aggregation_temporality`. As metrics of type Summary don't have an `aggregation_temporality` attribute, this attribute will be set to `AGGREGATION_TEMPORALITY_CUMULATIVE` for those metrics.
```

> The new metric that is created will be passed to all functions in the metrics statements list.  Function conditions will apply.

Reviewer 3: ```suggestion
The new metric that is created will be passed to all subsequent statements in the metrics statements list.
```
I think the last sentence should be a given since both Editors and conditions are available in any statement.

> The new metric that is created will be passed to all functions in the metrics statements list.  Function conditions will apply.

Reviewer 3: Could you add a test that the metric produced by this function will be executed in all subsequent statements? This is the first instance of a function adding additional metrics to my knowledge.

Reviewer: The existing summary metrics do this, but there's no such test for them either, so this is a good idea.
